sentinel/src/transitions/index.js

```diff
@@ -136,45 +113,31 @@ const processChange = (change, callback) => {
     });
 };
 
-const deleteReadDocs = (change, callback) => {
+const deleteReadDocs = change => {
   // we don't know if the deleted doc was a report or a message so
   // attempt to delete both
   const possibleReadDocIds = ['report', 'message'].map(
     type => `read:${type}:${change.id}`
   );
 
-  db.db.list((err, dbs) => {
-    if (err) {
-      return callback(err);
-    }
-
+  return db.allDbs().then(dbs => {
     const userDbs = dbs.filter(db => db.indexOf('medic-user-') === 0);
 
-    async.each(
-      userDbs,
-      (userDb, callback) => {
-        const metaDb = db.use(userDb);
-        metaDb.fetch({ keys: possibleReadDocIds }, (err, results) => {
-          if (err) {
-            return callback(err);
-          }
-          // find which of the possible ids was the right one
-          const row = results.rows.find(row => !!row.doc);
-          if (!row) {
-            return callback();
+    return Promise.all(userDbs.map(userDb => {
+      const metaDb = db.metaDb(userDb);
+      return metaDb.allDocs({ keys: possibleReadDocIds }).then(results => {
+        const row = results.rows.find(row => !row.error);
+        if (!row) {
+          return;
+        }
+
+        return metaDb.remove(row.id, row.value.rev).catch(err => {
+          if (err.status !== 404 && err.status !== 409) {
+            throw err;
```

Comment:
I think the comment about why we're ignoring 404s and 409s is still helpful.

Reply:
Done.